src/PhpSpec/Console/Application.php

```diff
@@ -57,9 +58,21 @@ public function getContainer()
      */
     public function doRun(InputInterface $input, OutputInterface $output)
     {
+        $helperSet = $this->getHelperSet();
         $this->container->set('console.input', $input);
         $this->container->set('console.output', $output);
-        $this->container->set('console.helper.dialog', $this->getHelperSet()->get('dialog'));
+        $this->container->setShared('console.prompter.factory', function($c) use ($helperSet) {
+            return new Factory(
+                $c->get('console.input'),
+                $c->get('console.output'),
+                $helperSet
+            );
+        });
+        if (!$this->container->isDefined('console.prompter')) {
+            $this->container->setShared('console.prompter', function($c) {
```

Comment:
Well, the factory currently relies on the helperset. Defining this one in the ContainerAssembler requires putting the helperset as a service on its own (which might be cleaner).

But the prompter itself should be defined in the assembler for sure.